app/grandchallenge/components/models.py

```diff
@@ -847,6 +912,10 @@ def __str__(self):
     def clean(self):
         super().clean()
 
+        if self.interface.kind == InterfaceKindChoices.SEGMENTATION:
+            self.interface._validate_pixel_values(
+                self.image_file, self._image_file.image_type
+            )
```

Comment:
This can go in the self.interface.is_image_kind block.

Won't this be called every time the CIV is saved? How have you ensured that this only runs in the Celery task on the acks-late-2xlarge queue?

Reply:
Thinking about this approach I think loading Image files in clean methods is asking for trouble. IMO the design should be revisited. Maybe something like this:

- A `segments` JSON field is added to `Image`
  - It is nullable (Default) or contains an array of the unique values in the image
  - We need to ensure that all of the values of an image are not dumped in there, so it has a maximum length, `MAXIMUM_SEGMENTS_LENGTH` (see below)
- When validation of a image is done anywhere the two sets of segments are compared
- `panimg` is responsible for all image interactions and getting the data so that we can validate images later
  - We extract the `segments` from each image there:
    - This is a array of numbers, or default `None`
    - We have to avoid doing this for large images so:
      - Segments should always be None for TIFF type images
      - Segments should always be None if the METAIO image type is not one of the Mask types (https://github.com/Kitware/MetaIO/blob/ffe3ce141c5a2394e40a0ecbe2a667cc0566baf5/src/metaTypes.h#L63-L94 - select the subset that are actually used as masks from there)
      - `panimg` adds a variable `MAXIMUM_SEGMENTS_LENGTH = 32`.
      - If the `number of segments > MAXIMUM_SEGMENTS_LENGTH` then `segments = None` for that image
- `MAXIMUM_SEGMENTS_LENGTH` is also added to the overlay schema 

This will save re-writing and guarding the clean method of the answer and CIV.